src/rt/runtime.rs

```diff
@@ -294,13 +204,11 @@ impl Machine {
                 runs = 0;
                 rt.quick_poll().unwrap();
 
-                if let Some(p) = self.processor.lock().as_mut() {
-                    if let Steal::Success(task) = p.steal_from_global(rt) {
-                        p.schedule(rt, task);
-                    }
-
-                    p.flush_slot(rt);
+                if let Steal::Success(task) = self.processor.steal_from_global(rt) {
+                    self.processor.schedule(rt, task);
```

Comment:
I never completely understood the proposed version of the reactor, so I don't know whether this change is accurate. Can you maybe share why the processor lock is no longer needed? I'm trying to understand better what's going on.

Reply:
Oh, It is I mistaked.
@stjepang created scheduler need to transfer machine's have processor. Because the number of threads changes.
https://github.com/async-rs/async-std/pull/631

However, the mechanism for changing the number of threads has been removed.So I thought I didn't need to lock the processor.

But, Worker haven't implemented Sync trait. Therefore, it was necessary to acquire a lock in order to handle the Processor in a thread-safe manner.

```
struct Processor {
    /// The local task queue.
    /// Not implemented Sync
    worker: Worker<Runnable>,

    /// Contains the next task to run as an optimization that skips the queue.
    slot: Cell<Option<Runnable>>,
}
```

I fixed it 3229111.